### PR TITLE
Upgrade/braze web sdk

### PR DESCRIPTION
**Pull request by aashishmalik**
## Description of the change

> Update Braze SDK from v2 to v4

## Type of change
- [-] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

- - -
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/617)

### CODE REVIEW

File: integrations/Braze/browser.js [hunk 1]
```diff
@@ -42,71 +43,33 @@ class Braze {
...
...eId initialize isPushBlocked isPushPermissionGranted isPushSupported logCardClick logCardDismissal logCardImpressions logContentCardImpressions logContentCardsDisplayed logCustomEvent logFeedDisplayed logInAppMessageButtonClick logInAppMessageClick logInAppMessageHtmlClick logInAppMessageImpression logPurchase openSession requestPushPermission removeAllSubscriptions removeSubscription requestContentCardsRefresh requestFeedRefresh requestImmediateDataFlush enableSDK isDisabled setLogger setSdkAuthenticationSignature addSdkMetadata disableSDK subscribeToContentCardsUpdates subscribeToFeedUpdates subscribeToInAppMessage subscribeToSdkAuthenticationFailures toggleLogging unregisterPush wipeData handleBrazeAction".split(" "),i=0;i<s.length;i++){for(var m=s[i],k=a.braze,l=m.split("."),j=0;j<l.length-1;j++)k=k[l[j]];k[l[j]]=(new Function("return function "+m.replace(/\./g,"_")+"(){window.brazeQueue.push(arguments); return true}"))()}window.braze.getCachedContentCards=function(){return new window.braze.ContentCards};window.braze.getCachedFeed=function(){return new window.braze.Feed};window.braze.getUser=function(){return new window.braze.User};(y=p.createElement(P)).type='text/javascript';
```
Comment (utsabc) [inline integrations/Braze/browser.js:57]: Can we move this large string to some constant variable in config.js file

**Review by utsabc — CHANGES_REQUESTED** [review]

File: integrations/Braze/browser.js [hunk 2]
```diff
@@ -42,71 +43,33 @@ class Braze {
...
-    window.appboy.display.automaticallyShowNewInAppMessages();
+    window.braze.automaticallyShowInAppMessages();
 
     const { userId } = this.analytics;
     // send userId if you have it https://js.appboycdn.com/web-sdk/latest/doc/module-appboy.html#.changeUser
-    if (userId) appboy.changeUser(userId);
-
-    window.appboy.openSession();
+    if (userId)
+      braze.changeUser(userId);
```
Comment (ItsSudip) [inline integrations/Braze/browser.js:71]:
```suggestion
    if (userId){
      braze.changeUser(userId);
     }
```

File: integrations/Braze/browser.js [hunk 3]
```diff
@@ -42,71 +43,33 @@ class Braze {
...
-      return window.appboy.ab.User.Genders.MALE;
+      return window.braze.User.Genders.MALE;
     if (otherGenders.indexOf(gender.toLowerCase()) > -1)
-      return window.appboy.ab.User.Genders.OTHER;
+      return window.braze.User.Genders.OTHER;
   }
 
   init() {
-    logger.debug("===in init Braze===");
+    logger.debug("===init Braze===");
```
Comment (ItsSudip) [inline integrations/Braze/browser.js:54]:
```suggestion
    logger.debug("===in init Braze===");
```

File: integrations/Braze/browser.js [hunk 4]
```diff
@@ -42,71 +43,33 @@ class Braze {
...
+      y.src='https://js.appboycdn.com/web-sdk/4.2/braze.min.js';
+      y.async=1;(b=p.getElementsByTagName(P)[0]).parentNode.insertBefore(y,b)
+    }(window,document,'script');
 
-    window.appboy.initialize(this.appKey, {
+    window.braze.initialize(this.appKey, {
       enableLogging: true,
       baseUrl: this.endPoint,
```
Comment (ItsSudip) [inline integrations/Braze/browser.js:64]: We are removing these two functionality. Are those not supported with the new SDK? If these can't be supported we have to remove these from our dashboard and docs.

**Review by ItsSudip — CHANGES_REQUESTED** [review]

File: integrations/Braze/browser.js (hunk 4 above)
Comment (aashishmalik) [reply integrations/Braze/browser.js:64]: which 2 functionalities ? only avatar support is removed

File: integrations/Braze/browser.js [hunk 5]
```diff
@@ -1,7 +1,8 @@
 /* eslint-disable class-methods-use-this */
+import { del } from "obj-case";
 import logger from "../../utils/logUtil";
-import { LOAD_ORIGIN } from "../ScriptLoader";
-import { NAME } from "./constants";
+import { LOAD_ORIGIN } from '../ScriptLoader';
```
Comment (ItsSudip) [inline integrations/Braze/browser.js:4]:
```suggestion
import { LOAD_ORIGIN } from "../ScriptLoader";
```

File: integrations/Braze/browser.js [hunk 6]
```diff
@@ -195,15 +190,15 @@ class Braze {
     });
 
     Object.keys(traits).forEach((key) => {
-      window.appboy.getUser().setCustomUserAttribute(key, traits[key]);
+      window.braze.getUser().setCustomUserAttribute(key, traits[key]);
```
Comment (ItsSudip) [inline integrations/Braze/browser.js:193]: For array they are using a different function [setCustomAttributeArray](https://www.braze.com/docs/developer_guide/platform_integration_guides/web/analytics/setting_custom_attributes/#implementation-examples-1:~:text=Setting%20a%20custom%20attribute%20with%20an%20array%20value).

**Review by ItsSudip — CHANGES_REQUESTED** [review]

**Review by ItsSudip — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**